aws/resource_aws_rds_cluster_activity_stream.go

```diff
@@ -0,0 +1,255 @@
+				Required: true,
+				ForceNew: true,
+			},
+			"apply_immediately": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"mode": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"sync",
+					"async",
+				}, false),
+			},
+			"kinesis_stream_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRDSClusterActivityStreamCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	resourceArn := d.Get("arn").(string)
+	applyImmediately := d.Get("apply_immediately").(bool)
+	kmsKeyId := d.Get("kms_key_id").(string)
+	mode := d.Get("mode").(string)
+
+	startActivityStreamInput := &rds.StartActivityStreamInput{
+		ResourceArn:      aws.String(resourceArn),
+		ApplyImmediately: aws.Bool(applyImmediately),
+		KmsKeyId:         aws.String(kmsKeyId),
+		Mode:             aws.String(mode),
+	}
+
+	log.Printf("[DEBUG] RDS Cluster start activity stream input: %s", startActivityStreamInput)
+
+	resp, err := conn.StartActivityStream(startActivityStreamInput)
+	if err != nil {
+		return fmt.Errorf("error creating RDS Cluster Activity Stream: %s", err)
+	}
+
+	log.Printf("[DEBUG]: RDS Cluster start activity stream response: %s", resp)
+
+	d.SetId(resourceArn)
+
+	err = resourceAwsRDSClusterActivityStreamWaitForStarted(d.Timeout(schema.TimeoutCreate), d.Id(), conn)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsRDSClusterActivityStreamRead(d, meta)
+}
+
+func resourceAwsRDSClusterActivityStreamRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	input := &rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Describing RDS Cluster: %s", input)
+	resp, err := conn.DescribeDBClusters(input)
+
+	if isAWSErr(err, rds.ErrCodeDBClusterNotFoundFault, "") {
+		log.Printf("[WARN] RDS Cluster (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error describing RDS Cluster (%s): %s", d.Id(), err)
+	}
+
+	if resp == nil {
+		return fmt.Errorf("error retrieving RDS cluster: empty response for: %s", input)
+	}
+
+	var dbc *rds.DBCluster
+	for _, c := range resp.DBClusters {
+		if aws.StringValue(c.DBClusterArn) == d.Id() {
+			dbc = c
+			break
+		}
+	}
+
+	if dbc == nil {
+		log.Printf("[WARN] RDS Cluster (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", dbc.DBClusterArn)
+	d.Set("kms_key_id", dbc.ActivityStreamKmsKeyId)
+	d.Set("kinesis_stream_name", dbc.ActivityStreamKinesisStreamName)
+	d.Set("mode", dbc.ActivityStreamMode)
+
+	return nil
+}
+
+func resourceAwsRDSClusterActivityStreamUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
i think it would be easier to omit update and set all fields as `ForceNew`

Reply:
After investigate about **apply_immediately**, i found when apply_immediately = false, rds.DescribeDBClusters API will not return ActivityStream associated attributes: ActivityStreamKmsKeyId, ActivityStreamKinesisStreamName and ActivityStreamMode. There only return ActivityStreamStatus.

So when apply_immediately = false, terraform state will persist empty string for ActivityStreamKmsKeyId, ActivityStreamKinesisStreamName and ActivityStreamMode. This will result in ForceNew fields when next applying, and terraform will recreate resource.

I think this condition terraform can't handling correctly. so i decide to delete **apply_immediately** in arguments and make **apply_immediately** always is true when creating resource.

I'm not sure if I understand it right